src/storage/rdb.cc

```diff
@@ -680,3 +679,323 @@ Status RDB::LoadRdb(uint32_t db_index, bool overwrite_exist_key) {
 
   return Status::OK();
 }
+
+Status RDB::Dump(const std::string &key, const RedisType type) {
+  unsigned char buf[2];
+  /* Serialize the object in an RDB-like format. It consist of an object type
+   * byte followed by the serialized object. This is understood by RESTORE. */
+  auto s = SaveObjectType(type);
+  if (!s.IsOK()) return {Status::RedisExecErr, s.Msg()};
+  s = SaveObject(key, type);
+  if (!s.IsOK()) return {Status::RedisExecErr, s.Msg()};
+
+  /* Write the footer, this is how it looks like:
+   * ----------------+---------------------+---------------+
+   * ... RDB payload | 2 bytes RDB version | 8 bytes CRC64 |
+   * ----------------+---------------------+---------------+
+   * RDB version and CRC are both in little endian.
+   */
+
+  /* RDB version */
+  buf[0] = MaxRDBVersion & 0xff;
+  buf[1] = (MaxRDBVersion >> 8) & 0xff;
+  s = stream_->Write((const char *)buf, 2);
+  if (!s.IsOK()) {
+    return {Status::RedisExecErr, s.Msg()};
+  }
+
+  /* CRC64 */
+  CHECK(stream_.get()->GetType() == String);
```

Comment:
Maybe just `dynamic_cast<RdbStringStream>(stream_.get()) != nullptr` ?

Reply:
`dynamic_cast` takes advantage of run-time type identification (RTTI), which may have an impact on performance. Although this scene is relatively simple, it may not have much impact. I'll modify it according to the comments.

Reply:
done